aiosmtpd/smtp.py

```diff
@@ -438,7 +438,12 @@ def smtp_DATA(self, arg):
                 'mail_options': self.mail_options,
                 'rcpt_options': self.rcpt_options,
             }
-        status = self.event_handler.process_message(*args, **kwargs)
+        if asyncio.iscoroutinefunction(self.event_handler.process_message):
+            kwargs['loop'] = self.loop
```

Comment:
I'm curious as to why you do this only for `iscoroutinefunction`s?  What's the harm in doing it unconditionally, or checking something else (e.g. that `self.loop is not None` or some such)?